authz/audit/stdout/stdout_logger.go

```diff
@@ -31,6 +31,9 @@ import (
 
 var grpcLogger = grpclog.Component("authz-audit")
 
+// Name is the string to identify this logger type in the registry
+const Name = "stdout_logger"
```

Comment:
Are all loggers going to be registered with a `_logger` suffix?  Isn't this redundant?  Is this standardized across languages or no?

Reply:
While I don't have strong preference on whether to keep this suffix, we are using the same name in [C++](https://github.com/grpc/grpc/blob/0d3ecd3e63cf3520d0d65b1f797cf54c1e97ac03/src/core/lib/security/authorization/stdout_logger.cc#L39) and it will be the same across languages such that the same authz policy can work everywhere.

Reply:
This was previously hardcoded [here](https://github.com/grpc/grpc-go/blob/a6e1acfc4420debbf347b0730e69d25b0a9c69ac/authz/audit/stdout/stdout_logger.go#L78)

I was just pulling it out to be an exported const so it could be looked up.

I'd be okay to change the name to `stdout`, what do you think @rockspore

Reply:
Oh wait..should this just not even be in the registry at all?  It seems like it's a built-in type that we support and not something that is supposed to be supported via the registry for generic loggers?

Reply:
I believe the intention with the built in types is that we place them in the registry for use - https://github.com/grpc/grpc-go/blob/a6e1acfc4420debbf347b0730e69d25b0a9c69ac/authz/audit/stdout/stdout_logger.go#L34-L38
@rockspore please check me here

Reply:
> I'd be okay to change the name to `stdout`, what do you think @rockspore

If there is no strong preference, I'm leaning towards keeping it as is so that we don't have to change what's already in C++ and the latest gRFC PR.

> Oh wait..should this just not even be in the registry at all?

We use the same registry for built-in loggers. We just pre-register them by importing the pkg here. The registry API probably needs to be prevent the built-in types from being overwritten.

Reply:
> We use the same registry for built-in loggers. We just pre-register them by importing the pkg here. The registry API probably needs to prevent the built-in types from being overwritten.

This doesn't sound right to me.

I think the built-in types should be hard-coded and not be present in the registry.  IIUC the registry is for the custom types that users can implement.  AFAICT there's no need to put the built-in types into the registry.  It buys us nothing and seems to have some downsides.

Reply:
I think this is an important but tangential-to-this-pr point to iron out.

@dfawley how would you feel about keeping this PR going as if the current behavior is the right behavior.
Then, we can resolve this separately, and if we change how the stdout logger is constructed we will update that in a separate PR.
As the audit logging is currently implemented, this is the correct way to do it.

Reply:
Right, we tried to stay similar to the LB policy for consistency

> But they're separate because we intend for them to be used through other pathways than xDS, right?

Yes, it can come through [here](https://github.com/grpc/grpc-go/blob/59134c303c31ba2ce65da8962c050918098f2a25/authz/grpc_authz_server_interceptors.go#L51) through `NewStatic`